arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (URL)/VectorTileLayerViewController.swift

```diff
@@ -20,59 +20,92 @@ import ArcGIS
+    /// The model array containing the layer options.
+    private let vectorTiledLayerItems: [(label: String, id: String)] = [
+        ("Mid-Century", "7675d44bb1e4428aa2c30a9b68f97822"),
+        ("Colored Pencil", "4cf7e1fb9f254dcda9c8fbadb15cf0f8"),
+        ("Newspaper", "dfb04de5f3144a80bc3f9f336228d24a"),
+        ("Nova", "75f4dfdff19e445395653121a95a85db"),
+        ("Night", "86f556a2d1fd468181855a35e344567f")
+    ]
     
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        //add the source code button item to the right of navigation bar
-        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["VectorTileLayerViewController"]
+        // add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = [
+            "VectorTileLayerViewController",
+            "OptionsTableViewController"
+        ]
         
-        //change width of the segmented control based on the device
-        if UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.pad {
-            self.segmentedControl.frame = self.segmentedControl.frame.insetBy(dx: -50, dy: 0)
-        }
-        
-        //Change text size of the segmented control to fit the names of maps
-        segmentedControl.setTitleTextAttributes([.font: UIFont.systemFont(ofSize: 10)], for: .normal)
+        /// The URL of the initial layer to display.
+        let url = arcGISURL(for: vectorTiledLayerItems.first!.id)
         
-        //create a vector tiled layer
-        let vectorTileLayer = AGSArcGISVectorTiledLayer(url: URL(string: midCenturyURLString)!)
-        //create a map and set the vector tiled layer as the basemap
+        // create a vector tiled layer
+        let vectorTileLayer = AGSArcGISVectorTiledLayer(url: url)
+        // create a map and set the vector tiled layer as the basemap
         let map = AGSMap(basemap: AGSBasemap(baseLayer: vectorTileLayer))
         
-        //assign the map to the map view
-        self.mapView.map = map
+        // assign the map to the map view
+        mapView.map = map
 
-        //center on Miami, Fl
-        self.mapView.setViewpointCenter(AGSPoint(x: -80.18, y: 25.778135, spatialReference: .wgs84()), scale: 150000)
+        // center on Miami, FL
+        mapView.setViewpointCenter(AGSPoint(x: -80.18, y: 25.778135, spatialReference: .wgs84()), scale: 150000)
 
     }
     
-    @IBAction func segmentedControlChanged(_ sender: UISegmentedControl) {
-        var urlString: String
-        switch sender.selectedSegmentIndex {
-        case 0:
-            urlString = midCenturyURLString
-        case 1:
-            urlString = coloredPencilURLString
-        case 2:
-            urlString = newsPaperURLString
-        case 3:
-            urlString = novaURLString
-        default:
-            urlString = nightURLString
+    private func arcGISURL(for id: String) -> URL {
```

Comment:
I would consider this a factory method, in which case the Swift API design guidelines suggest using a name like `makeArcGISURL(itemIdentifier:)`.

Reply:
Oh good call.